sdk/cognitivelanguage/ai-language-text/test/node/customTest.spec.ts

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { assertEnvironmentVariable, isPlaybackMode, Recorder } from "@azure-tools/test-recorder";
+import { AnalyzeBatchActionNames, AzureKeyCredential, TextAnalysisClient } from "../../src";
+import { matrix } from "@azure/test-utils";
+import { Context, Suite } from "mocha";
+import { AuthMethod, createClient, startRecorder } from "../public/utils/recordedClient";
+import createAuthoringClient, { TextAuthoringClient } from "@azure/ai-language-textauthoring";
+import {
+  cleanupCustomTestResource,
+  createCustomTestProject,
+} from "../public/utils/customTestHelpter";
+import { assertActionsResults } from "../public/utils/resultHelper";
+import { expectation1, expectation2, expectation4 } from "../public/expectations";
+matrix([["APIKey", "AAD"]] as const, async (authMethod: AuthMethod) => {
+  describe(`[${authMethod}] TextAnalysisClient`, function (this: Suite) {
+    let recorder: Recorder;
+    let client: TextAnalysisClient;
+    let authoringClient: TextAuthoringClient;
+
+    describe("analyzeBatch", function (this: Suite) {
+      const pollingInterval = isPlaybackMode() ? 0 : 2000;
+
+      describe("custom", function () {
+        let projectName: string;
+        let deploymentName: string;
+
+        before(function (this: Context) {
+          if (isPlaybackMode()) {
+            return;
+          }
+          // Create authoring client
+          const endpoint = assertEnvironmentVariable("ENDPOINT") || "<connection string>";
+          const apiKey = assertEnvironmentVariable("LANGUAGE_API_KEY") || "<api key>";
```

Comment:
```suggestion
          const endpoint = assertEnvironmentVariable("ENDPOINT");
          const apiKey = assertEnvironmentVariable("LANGUAGE_API_KEY");
```

Reply:
@harsha-nalluru can we load variables before the recorder is started?

Reply:
Yes, importing recorder automatically loads the environment variables from .env